### ISSUE TITLE
Flexibility around enum values.

### ISSUE THREAD

**Issue by FasterSpeeding** [nekokatt/hikari#182]
### Summary
The idea is to possibly avoid discord from breaking older versions of the lib when they instantaneously release new enum values with zero warning by trying to cast the value to the enum and falling back to using the raw value rather than letting the value error propagate leading to the event being dropped as this will effect stuff like cache reliability and may lead to events being dropped over arguably insignificant values.

### Problem
Discord doesn't know how versioning work.

### Ideal implementation
enum fields (excluding int flags as these don't use range check validation) will be a union of the enum or the raw value we get from discord (either string or integer) and will fall back to the raw value if they couldn't be casted.

### Checklist

- [x] I have searched the issue tracker and have made sure it's not a duplicate.
      If it is a follow up of another issue, I have specified it.

**Comment by davfsa** [comment]
This could be implemented along with #151. If this gets done, we should also remove the code that exists in the entity factory to account for this

**Comment by nekokatt** [comment]
If this is the case, we should probably just opt for a set of constant values rather than a proper enum.

I've already thought about this for stuff like Guild features.

```py
def namespace(cls):
    cls.__new__ = NotImplemented
    cls __init__ = NotImplemented

    def __init_subclass__(cls, **kwargs) -> NoReturn:
        raise TypeError("Cannot subclass namespace type")
    
    cls.__init_subclass__ = __init_subclass__
    return cls


```
@namespace
class GuildFeatures:
    PUBLIC = "public"
    DISCOVERABLE = "discoverable"
```

Makes me question if there is even any benefit in documenting what values these can be if they are volatile.

**Comment by nekokatt** [comment]
Will be able to be fixed as part of #208 as it is in the same place. I will close this and document the issue there to keep them together.
